skbase/lookup/_lookup.py

```diff
@@ -194,28 +203,36 @@ def _filter_by_tags(
     """
     if tag_filter is None:
         return True
+
     if hasattr(klass, "get_class_tags"):
         klass_tags = klass.get_class_tags()
+
+    else:
+        # If the object doesn't have tag interface it can't have the tags in tag_filter
+        return False
```

Comment:
interesting question: the user expected behaviour may differ in this case, depending on whether a default or a non-default is being filtered by.

For simplicity, I agree with this, but I wonder whether there is a nicer solution. Not a blocking comment.

Reply:
@fkiraly what do you mean by default or non-default?

I did waffle in this, but felt this was safer of two simple options (other being classes without tag interface are not filtered at all).

Reply:
ok, makes sense, especially since it would otherwise pick up all (python) objects.